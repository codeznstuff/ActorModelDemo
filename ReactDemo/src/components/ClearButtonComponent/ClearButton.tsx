import React from 'react';
import styles from './ClearButton.module.scss';
import { clearButtonProps } from '../../@types/constants';

export const ClearButton = (props: clearButtonProps) => (
    <div className={styles.clear_btn} onClick={() => props.handleClear()} data-testid="clearButton">
        <u>C</u>LEAR
    </div>
)

export default ClearButton