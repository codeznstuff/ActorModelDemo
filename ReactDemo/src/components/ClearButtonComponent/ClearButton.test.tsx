import React from 'react';
import { cleanup } from '@testing-library/react'
import { ClearButton } from './ClearButton';
import Enzyme, { shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe("ClearButtonComponent", () => {
    let button: ShallowWrapper;
    let mockClick: jest.Mock;
    beforeEach(() => {
        mockClick = jest.fn();
        button = shallow(<ClearButton handleClear={mockClick} />);
    });
    it('Button Snapshot should match', () => {
        expect(button.debug()).toMatchSnapshot();
    })
    it('Button should have text equal to CLEAR', () => {
        expect(button.text()).toBe("CLEAR");
    });
    it('Should call function on button click', async () => {
        button.simulate('click')
        expect(mockClick).toBeCalled();
    });
    afterEach(cleanup);
})

