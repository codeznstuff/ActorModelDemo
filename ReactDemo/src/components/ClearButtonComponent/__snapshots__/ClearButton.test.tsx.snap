// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearButtonComponent Button Snapshot should match 1`] = `
"<div className=\\"clear_btn\\" onClick={[Function: onClick]} data-testid=\\"clearButton\\">
  <u>
    C
  </u>
  LEAR
</div>"
`;
